### PR TITLE
Fix gRPC examples that are failing.

### PR DESCRIPTION
**Pull request by klustria**
1. Use built-in JsobBMarshaller for the gRPC request/response that needs it by setting it as the custom marshaller.
2. Update pom.xml for microservices example because the contents are already obsolete.
3. Enhance the READMEs across all the examples.
4. Update config for security-outbound example to add outbound entry so that the security context can properly be propagated from gRPC to Rest and vice-versa.
5. Remove grpc.marshaller.java.enabled from the config as it is not used anymore.
6. Replace policy with policy-validator as the former is an invalid ABAC attribute.

### CODE REVIEW

**Comment by klustria** [comment]
This should fix [GRPC examples are failing #4576](https://github.com/oracle/helidon/issues/4576)

File: examples/grpc/common/src/main/java/io/helidon/grpc/examples/common/JavaMarshaller.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import io.grpc.MethodDescriptor;
+import jakarta.inject.Named;
+
+/**
+ * An implementation of a gRPC {@link MethodDescriptor.Marshaller} that
+ * uses Java serialization for testing.
+ *
+ * @param <T>  the type of value to be marshalled
+ */
+public class JavaMarshaller<T>
```
Comment (tomas-langer) [inline examples/grpc/common/src/main/java/io/helidon/grpc/examples/common/JavaMarshaller.java:42]: Do we really need to use a Java marshaller? We actually wanted to remove any serialization from Helidon, and this is just re-introducing it into examples. 
We should use a different serialization approach in examples (proto, json) instead

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: examples/grpc/common/src/main/java/io/helidon/grpc/examples/common/JavaMarshaller.java (hunk 1 above)
Comment (klustria) [reply examples/grpc/common/src/main/java/io/helidon/grpc/examples/common/JavaMarshaller.java:42]: Removed and replaced with built-in JsonBMarshaller

**Review by arjav-desai — APPROVED** [review]
LGTM

**Pull request merged**